compiler/rustc_mir_transform/src/check_alignment.rs

```diff
@@ -75,6 +75,14 @@ struct PointerFinder<'tcx, 'a> {
 }
 
 impl<'tcx, 'a> Visitor<'tcx> for PointerFinder<'tcx, 'a> {
+    fn visit_rvalue(&mut self, rvalue: &Rvalue<'tcx>, location: Location) {
+        if let Rvalue::AddressOf(..) = rvalue {
+            // Ignore dereferences inside of an AddressOf
```

Comment:
Would be good to have a comment here that these derefs (if unaligned) *are* UB according to the reference, but we choose to skip them here because they there are stull debates wrt whether we want to relax this UB later or not.

Also  (while I am here), IMO `if let` without any binder is better replaced by `if matches!`.

Reply:
@thomcc would you be willing to create a meeting proposal at https://github.com/rust-lang/opsem-team/ ?

Reply:
Done: https://github.com/rust-lang/opsem-team/issues/9